src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1440,6 +1440,15 @@ private AuthenticationResult localFlow(CallbackHandler callbackHandle,
             } catch (AuthenticationException authenticationException) {
                 callbackHandle.onError(authenticationException);
                 return null;
+            } catch (HttpResponseException httpResponseException) {
+                // Return back the http error for silent request
+                // Try interactive flow for non-silent request.
+                if (request.isSilent()) {
+                    final AuthenticationException authenticationException = new AuthenticationException(ADALError.AUTH_REFRESH_FAILED_PROMPT_NOT_ALLOWED, 
```

Comment:
AUTH_REFRESH_FAILE_PROMPT_NOT_ALLOWED is a general error code we currently return back to user indicating "silent request failed, you need to start interactive flow". 

The thing that's hard for us is, the above error code is what we return today, and an agreement we have with our consumers that it basically means you need interactive flow. Changing the error code will break the existing agreement we have.